doc/manual/23-wal_streaming.en.md

```diff
@@ -104,6 +104,45 @@ create the replication slot by setting:
 create_slot = auto
 ```
 
+### Streaming WALs and backups from different hosts (Barman 3.10.0 and later)
+
+Barman uses the connection info defined in `streaming_conninfo` when creating
+`pg_receivewal` processes to stream WAL segments and uses `conninfo` when
+checking the status of replication slots. Because `conninfo` and
+`streaming_conninfo` are also used when taking backups this default
+configuration forces Barman to stream WALs and take backups from the same host.
+
+If an alternative configuration is required, such as backups being sourced from
+a standby with WALs being streamed from the primary, then this can be achieved
+using the following options:
+
+- `wal_streaming_conninfo`: A connection string which Barman will use instead
+   of `streaming_conninfo` when receiving WAL segments via the streaming
+  replication protocol and when checking the status of the replication slot
+  used for receiving WALs.
+- `wal_conninfo`: A connection string which Barman will use instead of
+  `wal_streaming_conninfo` when checking the status of the replication slot
```

Comment:
No it is supposed to be `wal_streaming_conninfo` - or at least, that was my intention however maybe we need to rethink it...

The idea is that `wal_streaming_conninfo` is expected to be a conninfo string which can make streaming replication connections and also make SQL connections with monitoring privileges, so if only `conninfo`, `streaming_conninfo` and `wal_streaming_conninfo` are defined, then Barman uses `wal_streaming_conninfo` for receiving WALs via the replication and Barman also uses `wal_streaming_conninfo` for querying the status of the replication slot and other streaming related checks.

For users who do not want one conninfo string having permissions to make streaming replication connections and check monitoring statistics, they can set `wal_conninfo` and this will be used instead of `wal_streaming_conninfo` for checking replication slots and related queries.

This feels over complicated, but if we were to have `wal_conninfo` override `conninfo` then Barman would use `conninfo` with `wal_streaming_conninfo` and the likelihood is these would point to different hosts and therefore give misleading information (and probably fail for reasons like not being able to find the replication slot).

I don't know if we can make it clearer in the docs about how these are intended to be used, or whether perhaps there is a simpler scheme that I am not seeing which would be more intuitive - can you think of a better approach here?

Reply:
Maybe this is a bit clearer?

```
- `wal_conninfo`: An optional connection string specifically for monitoring
  WAL streaming status and performing related checks. If set, Barman will use
  this instead of `wal_streaming_conninfo` when checking the status of the
  replication slot.
```

Reply:
Oh, that was an oversight from my side. IAC I guess your new descriptions is indeed better. Thanks for expanding!